File: computer-based-test/app/Models/latihan.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class latihan extends Model
{

    protected $table = 'latihan'; 

    protected $primaryKey = 'id_latihan';

    protected $fillable = [
        'id_latihan',
        'soal_latihan',
        'acak',
        'status_jawaban',
        'grade',
        'id_guru',
    ];

    public function guru(){
        return $this->belongsTo(guru::class);
    }

    public function soal(){
        return $this->hasMany(soal::class);
    }
}

File: cbt-api_before/entity/latihan.go
package entity

import "time"

type Latihan struct {
    IdLatihan     uint64    `gorm:"primary_key;autoIncrement" json:"id_latihan"`
    SoalLatihan   string    `gorm:"type:varchar(255);not null" json:"soal_latihan"`
    Acak          string    `gorm:"type:enum('Aktif', 'Tidak Aktif');not null" json:"acak"`
    StatusJawaban string    `gorm:"type:enum('Aktif', 'Tidak Aktif');not null" json:"status_jawaban"`
    Grade         float64   `gorm:"type:double" json:"grade"`
    IdGuru        uint64    `gorm:"type:bigint(20) unsigned" json:"id_guru"`
    CreatedAt        time.Time `gorm:"type:timestamp;default:CURRENT_TIMESTAMP" json:"created_at"`
    UpdatedAt        time.Time `gorm:"type:timestamp;default:CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP" json:"updated_at"`

    Guru    Guru    `gorm:"foreignkey:IdGuru;constraint:onUpdate:CASCADE, onDelete:CASCADE" json:"guru"`
}
